src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultViewModel.swift

```diff
@@ -91,6 +93,9 @@ class ExposureSubmissionTestResultViewModel {
 	func deleteTest() {
 		exposureSubmissionService.deleteTest()
 		onTestDeleted()
+		
+		// Update warn others model
+		self.warnOthers.reset()
```

Comment:
I'm wondering, could maybe all warn others calls be put into the exposure submission service? Maybe apart from the initial scheduling the the user actually sees the test result for the first time, that can only be ensured in the view controller. But everything else, like this reset and the cancelNotifications on successful submit could also be handled in the exposure submission coordinator. That way it would not have to be spread around various view controllers and not be considered for other paths that may be added in the future where e.g. `exposureSubmissionService.deleteTest()` is called.

Reply:
We'll take this into the refactoring rework next sprint